source/functions.rst

```diff
@@ -1,191 +1,186 @@
 Functions
 *********
 
-You can think of functions as actions, verbs, or commands 
-and you can think of parameters as adverbs: 'run, quickly'
+A functions is some code which has been given a name and which can be used
+wherever you need it in your program. When we use a function we say we are
+"calling" it.
 
-Functions are special objects that contain code.
+Some examples of functions we've already used are ``print``, ``int`` and
+``type``.
 
-When you call them, using special syntax `()`, you execute the
-code they contain.
+    >>> print("hello") hello
 
-::
+The ``print`` function simply prints out what is between the brackets. We use
+one value, or "argument", when calling ``print``, but some functions can be
+called with zero arguments, for example the ``exit`` function:
 
-    >>> turtle.forward             # functions have names
-    <function turtle.forward>
-    >>> turtle.forward(10)         # actioned by use of '()'s
+    >>> exit()
 
+Other functions require more than one argument, for example the ``max``
+function:
 
-All turtle instructions are examples of calling functions attached 
-to the turtle object.
-
-`print` is another function::
-    
-    >>> print('hello')
-
-`print` simply prints its parameter to the console.
+    >>> max(1, 2)
+    2
+    >>> max(-1000, 1000, 0)
+    1000
 
 .. tip::
 
-    Functions and methods are very similar. 
-    Methods exist on objects however functions stand alone.
-
+   Like everything else in Python, functions are objects. Function objects
+   contain code which you run using the brackets containing whatever arguments
+   that function needs.
 
-Function objects
-================
 
-A function like everything in Python is an object. Function objects are different in that they contain blocks of code.
+Defining and using functions
+============================
 
-Functions help in letting programmers organise and reuse code. They help create new abstractions.
+Functions help you to organise your code better. You can split up complicated
+tasks into several smaller, simpler ones. And they allow you to reuse code
+without repeating it.
 
-Function objects have names. The name is assigned at the same time you define a function.
+Definining functions
+--------------------
 
-defining
---------
+Every function has to be "defined" before it can be used. To define a function
+you use ``def``:
 
-Creating function objects requires special syntax::
-
-    >>> def my_function():
-    ...     print('hello')    # Note 4 space indentation
+    >>> def say_hello(name):
+    ...     print("hello " + name + "!")
     ...
-    >>> type(my_function)
-    function
-
-The ``def`` keyword is followed by the function object name, followed by () and then a colon. 
```

Comment:
whoops. You have it here: https://github.com/gu-lseg/intro-to-programming/pull/33/files#diff-900e076ad8ebe3c8ab60140c776c7af4R5. 👍